tutorials/preprocessing/plot_15_handling_bad_channels.py

```diff
@@ -0,0 +1,209 @@
+# analysis stream without actually deleting the data in those channels, by
+# keeping track of the bad channel indices in a list and looking at that list
+# when doing analysis or plotting tasks. The list of bad channels is stored in
+# the `'bads'` field of the :class:`~mne.Info` object that is attached to
+# :class:`~mne.io.Raw`, :class:`~mne.Epochs`, and :class:`~mne.Evoked` objects.
+
+print(raw.info['bads'])
+
+###############################################################################
+# Here you can see that the :file:`.fif` file we loaded from disk must have
+# been keeping track of channels marked as "bad" — which is good news, because
+# it means any changes we make to the list of bad channels will be preserved
+# if we save our data at intermediate stages and re-load it later. In the case
+# of the example data, you can see why those channels were marked bad by
+# using the standard :meth:`~mne.io.Raw.plot` method; to make it easier to see
+# we'll use the :func:`~mne.pick_channels_regexp` function to narrow down which
+# channels we see:
+
+# sphinx_gallery_thumbnail_number = 2
+for pattern in (r'MEG 24..', r'EEG 05.'):
+    picks = mne.pick_channels_regexp(raw.ch_names, regexp=pattern)
+    raw.plot(order=picks, n_channels=len(picks))
+
+###############################################################################
+# Notice first of all that the channels marked as "bad" are plotted in a light
+# gray color in a layer behind the other channels, to make it easy to
+# distinguish them from "good" channels. The plots make it clear that `EEG 053`
+# is not picking up scalp potentials at all, and `MEG 2443` looks like it's
+# picking up a lot *more* than its neighbors — its signal is at least an order
+# of magnitude greater than the other MEG channels. Regardless of what is
+# causing the problems, neither are realistic reflections of the
+# electrophysiological signals we expect, so it makes sense to exclude them.
+#
+# If you want to change which channels are marked as bad, you can edit
+# ``raw.info['bads']`` directly; it's an ordinary Python :class:`list` so the
+# usual list methods will work:
+
+original_bads = raw.info['bads']
+raw.info['bads'].append('EEG 050')               # add a single channel
+raw.info['bads'].extend(['EEG 051', 'EEG 052'])  # add a list of channels
+bad_chan = raw.info['bads'].pop(-1)  # remove the last entry in the list
+raw.info['bads'] = original_bads     # change the whole list at once
+
+###############################################################################
+# .. sidebar:: Blocking execution
+#
+#     If you want to build an interactive bad-channel-marking step into an
+#     analysis script, be sure to include the parameter ``block=True`` in your
+#     call to ``raw.plot()`` or ``epochs.plot()``. This will pause the script
+#     while the plot is open, giving you time to mark bad channels before
+#     subsequent analysis or plotting steps are executed.
```

Comment:
Perhaps consider adding that this can be particularly convenient when looping over subjects.